src/AlignLoads.cpp

```diff
@@ -0,0 +1,174 @@
+            op->param.defined() ? op->param.host_alignment() : required_alignment;
+        if (base_alignment % required_alignment == 0) {
+            // We know the base is aligned. Try to find out the offset
+            // of the ramp base from an aligned offset.
+            known_alignment = reduce_expr_modulo(ramp->base, native_lanes, &aligned_offset,
+                                                 alignment_info);
+        }
+
+        int stride = static_cast<int>(*const_stride);
+        if (stride != 1) {
+            internal_assert(stride >= 0);
+
+            // If we know the offset of this strided load is smaller
+            // than the stride, we can just make the load aligned now
+            // without requiring more vectors from the dense
+            // load. This makes loads like f(2*x + 1) into an aligned
+            // load of double length, with a single shuffle.
+            int shift = known_alignment && aligned_offset < stride ? aligned_offset : 0;
+
+            // Load a dense vector covering all of the addresses in the load.
+            Expr dense_base = simplify(ramp->base - shift);
+            Expr dense_index = Ramp::make(dense_base, 1, (lanes - 1)*stride + 1 + shift);
+            Expr dense = make_load(op, dense_index);
+
+            // Shuffle the dense load.
+            expr = slice_vector(dense, shift, stride, lanes);
+            return;
+        }
+
+        // We now have a dense vector load to deal with.
+        internal_assert(stride == 1);
+        if (lanes < native_lanes) {
+            // This load is smaller than a native vector. Load a
+            // native vector.
+            Expr native_load = make_load(op, Ramp::make(ramp->base, 1, native_lanes));
+
+            // Slice the native load.
+            expr = slice_vector(native_load, 0, 1, lanes);
+            return;
+        } else if (lanes > native_lanes) {
+            // This load is larger than a native vector. Load native
+            // vectors, and concatenate the results.
+            vector<Expr> slices;
+            for (int i = 0; i < lanes; i += native_lanes) {
+                int slice_lanes = std::min(native_lanes, lanes - i);
+                Expr slice_base = simplify(ramp->base + i);
+                slices.push_back(make_load(op, Ramp::make(slice_base, 1, slice_lanes)));
+            }
+            expr = Call::make(op->type, Call::concat_vectors, slices, Call::PureIntrinsic);
+            return;
+        } else if (known_alignment && aligned_offset != 0) {
+            // We know the offset of this load from an aligned
+            // address. Rewrite this is an aligned load of two
+            // native vectors, followed by a shuffle.
+            Expr aligned_base = simplify(ramp->base - aligned_offset);
+            Expr aligned_load = make_load(op, Ramp::make(aligned_base, 1, lanes*2));
+
+            expr = slice_vector(aligned_load, aligned_offset, 1, lanes);
+            return;
+        }
```

Comment:
I prefer to remove the elses. This whole function is structured as a series of stages where it can early-out at each step, I don't like how it looks if we have the function structured like that, until the last if, which uses 'else' to handle the early-out.